src/v0/destinations/am/utils.js

```diff
@@ -82,6 +82,32 @@ const getEventId = (payload, sourceKey) => {
   return undefined;
 };
 
+/**
+ * generates the unsetObject and returns it 
+ * @param {*} message 
+ * @returns 
+ * 
+ * Example message = {
+    integrations: {
+      Amplitude: { fieldsToUnset: ['Unset1', 'Unset2'] },
+      All: true,
+    },
+  };
+  return unsetObj = {
+        "Unset1": "-",
+        "Unset2": "-"
+      }
+  AM docs: https://www.docs.developers.amplitude.com/analytics/apis/http-v2-api/#keys-for-the-event-argument:~:text=exceed%2040%20layers.-,user_properties,-Optional.%20Object.%20A
+ */
+const getUnsetObj = (message) => {
+  const fieldsToUnset = get(message, 'integrations.Amplitude.fieldsToUnset');
+  let unsetObject;
+  if (fieldsToUnset && Array.isArray(fieldsToUnset)) {
```

Comment:
Just `Array.isArray(fieldsToUnset)` should suffice

Reply:
Can you once confirm ?

Reply:
done
![Uploading Screenshot 2023-12-22 at 3.29.59 PM.png…]()